src/ultrasonic/main.c

```diff
@@ -1,13 +1,56 @@
 /**
  * main (entrypoint) implementation
  */
+#include "timercontrol.h"
 #include "init.h"
+#include "pinmap.h"
+#include <avr/io.h>
+#include <util/delay.h>
+#include <avr/interrupt.h>
+#include "stdint.h"
+#include "bit_utils.h"
+#include "globals.h"
+
+
+float getDistance(int trigPin, int frequency, uint8_t flag) {
```

Comment:
is `flag` going to some sort of atomic global? If so are you tying to modify in place? If not, why does L32 exist?